tests/integration-tests/utils.py

```diff
@@ -357,12 +384,7 @@ def remove_keys_from_known_hosts(hostname, host_keys_file, env):
 
 def get_instance_info(instance_type, region_name=None):
```

Comment:
what I don't like is that this function relies on a global state (variable) which needs to be set in the InstanceTypesData.
One alternative which I probably like more is to convert this into a fixture and have something like:

```
@pytest.fixture(scope="class")
def instance_type_info(region, request):
    additional_data = request.config.getoption("instance_types_data")
    def _get_instance_type_info(instance_type):
       # here implement the logic to return the instance_type_info

   return _get_instance_type_info
```

The test that requires this will just inject it as a fixture rather than importing the funciton
```
def test_function(instance_type_info):
   instance_type_info(instance)
```

Reply:
I agree it should be better. However this function is called from other fixtures, so I would like to clarify the approach here.
Let's talk about this.